### PR TITLE
Docs: Change design of search box #688

### LINKED ISSUE #688: Docs: Change design of search box
# Tasks

- [x] Docs画面左上の検索ボックスのデザインを、Tailwindのサイトと同様のものにする

# Writing the Issue

- [x] Write each task for this issue using checkboxes
- [x] Write all verbal instructions as tasks

# Before Working on the Issue

- [x] Assign yourself
- [x] Share your screen on Discord
- [x] Write which issue you are starting on Slack
- [x] Git: Fetch the latest main
- [x] Git: Create a branch with the name of the issue

# Pre-PR Checks

- [x] Insure that changes include only what is necessary for this PR
- [x] Implement necessary Unit tests
- [x] Implement necessary E2E tests
- [x] Perform functionality checks

### PR DESCRIPTION
**Pull request by sakata0411**
closes #688 

# For the Submitter

## Tasks

- [x] Make sure "closes (issue number)" is written close to the top
- [x] Add reviewers
- [x] Make sure CI completes successfully
- [x] On Slack, mention the PR reviewer, write your thoughts on the issue, and inform them that the PR has been created

# For the Reviewer

## Before Reviewing

- [x] Write which PR you are reviewing on Slack

## Checks

- [x] Each task for the related issue are written and in checkboxes
- [x] The branch name is close to the issue name
- [x] CI completes successfully
- [x] Changes include only what is necessary for this PR
- [x] Necessary unit tests are implemented
- [x] Necessary E2E tests are implemented
- [x] Perform functionality checks
- [x] If there are any problems, request revisions from the PR submitter and mention them on Slack

## After Reviewing

- [x] Leave an approval comment (e.g. LGTM)
- [x] On Slack, mention the PR submitter and a project admin, write which PR you have reviewed, and ask them to merge the PR

### CODE REVIEW

File: src/routes/docs/side-bar.svelte [hunk 1]
```diff
@@ -3,15 +3,28 @@
...
 	import type { Section } from '$lib/docs/markdown'
 	import { createEventDispatcher } from 'svelte'
+	import { onMount } from 'svelte'
+	import { ModifierKey } from '$lib/view/modifier_key'
 
 	export let sections: Section[]
 	export let search_bar_enabled = true
 
+	let view_shortcut_key = ''
+	let cursor_on_search = false
```
Comment (milkyskies) [inline src/routes/docs/side-bar.svelte:13]: これどこで使ってますか？

File: src/routes/docs/side-bar.svelte [hunk 2]
```diff
@@ -3,15 +3,28 @@
...
 
 	function on_search_button_click(): void {
 		dispatch('show_search_modale')
 	}
+
+	onMount(() => {
+		const modifier_key = new ModifierKey()
+
+		view_shortcut_key = modifier_key.get_control_or_command_symbol()
+	})
```
Comment (milkyskies) [inline src/routes/docs/side-bar.svelte:25]: - [ ] 関数に抽出

File: src/routes/docs/side-bar.svelte [hunk 3]
```diff
@@ -42,9 +55,19 @@
...
+			class="flex glass-panel gap-3 items-center mt-8 w-full rounded-md bg-slate-900/90 hover:bg-slate-500/75 transition-all duration-150"
+			on:click={on_search_button_click}
+			on:mousemove={() => (cursor_on_search = true)}
+			on:mouseleave={() => (cursor_on_search = false)}
+		>
 			<div class="h-5"><SearchIcon /></div>
-			Search
+			<div class="flex w-full justify-between pr-2">
+				<div class="flex">Search</div>
+				{#if view_shortcut_key.length > 0}
```
Comment (milkyskies) [inline src/routes/docs/side-bar.svelte:67]:
```suggestion
				{#if view_shortcut_key}
```

**Review by milkyskies — CHANGES_REQUESTED** [review]

File: src/routes/docs/side-bar.svelte (hunk 1 above)
Comment (sakata0411) [reply src/routes/docs/side-bar.svelte]: 消し忘れだったため関連部分も含めて削除しました

**Review by milkyskies — APPROVED** [review]
LGTM

**Pull request merged**